ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -200,6 +203,52 @@ public boolean onTouchEvent(MotionEvent ev) {
     return true;
   }
 
+  @Override
+  public boolean dispatchKeyEvent(KeyEvent ev) {
+    if (mReactInstanceManager == null || !mIsAttachedToInstance ||
+      mReactInstanceManager.getCurrentReactContext() == null) {
+      FLog.w(
+        ReactConstants.TAG,
+        "Unable to handle key event as the catalyst instance has not been attached");
+      return super.dispatchKeyEvent(ev);
+    }
+    mAndroidTVRootViewHelper.handleKeyEvent(ev, getDeviceEventEmitter());
+    return super.dispatchKeyEvent(ev);
+  }
+
+  @Override
+  protected void onFocusChanged(boolean gainFocus, int direction, Rect previouslyFocusedRect) {
+    if (mReactInstanceManager == null || !mIsAttachedToInstance ||
+      mReactInstanceManager.getCurrentReactContext() == null) {
+      FLog.w(
+        ReactConstants.TAG,
+        "Unable to handle focus changed event as the catalyst instance has not been attached");
+      super.onFocusChanged(gainFocus, direction, previouslyFocusedRect);
+      return;
+    }
+    mAndroidTVRootViewHelper.clearFocus(getDeviceEventEmitter());
+    super.onFocusChanged(gainFocus, direction, previouslyFocusedRect);
+  }
+
+  @Override
+  public void requestChildFocus(View child, View focused) {
+    if (mReactInstanceManager == null || !mIsAttachedToInstance ||
+      mReactInstanceManager.getCurrentReactContext() == null) {
+      FLog.w(
+        ReactConstants.TAG,
+        "Unable to handle child focus changed event as the catalyst instance has not been attached");
+      super.requestChildFocus(child, focused);
+      return;
+    }
+    mAndroidTVRootViewHelper.onFocusChanged(focused, getDeviceEventEmitter());
+    super.requestChildFocus(child, focused);
+  }
+
+  private DeviceEventManagerModule.RCTDeviceEventEmitter getDeviceEventEmitter() {
```

Comment:
@kmagiera Do you think this is a good idea to separate this (and maybe also the part with null checking catalyst instance) to a separate method, since it is repeated multiple times in these proxy methods?

Reply:
I don't think we will save much by doing this. Not worth changing at this moment